File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DocumentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    owner_id: int
    original_filename: str
    content_type: str | None
    size_bytes: int
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/document.ts
export type IndexingStatus =
  | "PROCESSING"
  | "COMPLETED"
  | "FAILED";

export type DocumentType =
  | "PDF"
  | "DOCX"
  | "TXT";

export interface Document {
  id: number;
  owner_id: number;
  original_filename: string;
  content_type: string | null;
  size_bytes: number;
  created_at: string;
  updated_at: string;
}

export interface DocumentVersion {
  id: number;
  document_id: number;
  version_number: number;
  updated_at: string;
  is_current: boolean;
  available: boolean;
}